dataset_builders/pie/comagc/comagc.py

```diff
@@ -0,0 +1,198 @@
+
+
+def example_to_document(example) -> ComagcDocument:
+    metadata = {
+        "cancer_type": example["cancer_type"],
+        "annotation": {
+            "CGE": example["CGE"],
+            "CCS": example["CCS"],
+            "PT": example["PT"],
+            "IGE": example["IGE"],
+        },
+        "expression_change_keywords": [
+            example["expression_change_keyword_1"],
+            example["expression_change_keyword_2"],
+        ],
+    }
+
+    doc = ComagcDocument(
+        text=example["sentence"],
+        id=example["pmid"],
+        metadata=metadata,
+    )
+    # entity name is the text of the entity (between the start and end positions)
+    head = LabeledSpan(
+        start=example["gene"]["pos"][0],
+        end=example["gene"]["pos"][1] + 1,
+        label="GENE",
+    )
+    tail = LabeledSpan(
+        start=example["cancer"]["pos"][0],
+        end=example["cancer"]["pos"][1] + 1,
+        label="CANCER",
+    )
+    doc.entities.extend([head, tail])
+
+    relation = BinaryRelation(head=head, tail=tail, label=get_relation(example))
+    doc.relations.append(relation)
+    return doc
+
+
+def document_to_example(doc: ComagcDocument) -> Dict[str, Any]:
+    gene = {
+        "name": doc.text[doc.entities[0].start : doc.entities[0].end],
+        "pos": [doc.entities[0].start, doc.entities[0].end - 1],
+    }
+    cancer = {
+        "name": doc.text[doc.entities[1].start : doc.entities[1].end],
+        "pos": [doc.entities[1].start, doc.entities[1].end - 1],
+    }
+
+    return {
+        "pmid": doc.id,
+        "sentence": doc.text,
+        "cancer_type": doc.metadata["cancer_type"],
+        "gene": gene,
+        "cancer": cancer,
+        "CGE": doc.metadata["annotation"]["CGE"],
+        "CCS": doc.metadata["annotation"]["CCS"],
+        "PT": doc.metadata["annotation"]["PT"],
+        "IGE": doc.metadata["annotation"]["IGE"],
+        "expression_change_keyword_1": doc.metadata["expression_change_keywords"][0],
+        "expression_change_keyword_2": doc.metadata["expression_change_keywords"][1],
+    }
+
+
+class Comagc(ArrowBasedBuilder):
+    DOCUMENT_TYPE = ComagcDocument
+    BASE_DATASET_PATH = "DFKI-SLT/CoMAGC"
+    BASE_DATASET_REVISION = "8e2950b8a3967c2f45de86f60dd5c8ccb9ad3815"
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            version=datasets.Version("1.0.0"),
+            description="CoMAGC dataset",
+        )
+    ]
+
+    @property
+    def document_converters(self):
+        return {
+            TextDocumentWithLabeledSpansAndBinaryRelations: {
+                "entities": "labeled_spans",
+                "relations": "binary_relations",
+            }
+        }
+
+    def _generate_document(self, example, **kwargs):
+        return example_to_document(example)
+
+    def _generate_example(self, document: Document, **kwargs) -> Dict[str, Any]:
+        return document_to_example(document)
+
+
+def get_relation(example):
```

Comment:
add type annotations (primarily important is the return type). From the naming I assume this returns a `BinaryRelation`?

Reply:
Adjusted accordingly, changed to: `def get_relation_label(example: Dict) -> str:`

Reply:
Just commited